endpoints/server.go

```diff
@@ -129,23 +133,39 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Initialize RPC method response and call method's function
-	resp := reflect.New(methodSpec.RespType)
-	errValue := methodSpec.method.Func.Call([]reflect.Value{
-		serviceSpec.rcvr,
-		reflect.ValueOf(r),
-		req,
-		resp,
-	})
+	var respValue, errValue reflect.Value
+
+	// Construct arguments for the method call
+	args := make([]reflect.Value, 0, 4)
```

Comment:
But then wouldn't it be even simpler to replace this and the next lines with:
`args := []reflect.Value{serviceSpec.rcvr}` ?